lib/AppConfig.php

```diff
@@ -140,10 +163,28 @@ public function getAppValue($key) {
 	 * @param string $key
 	 * @param string $value
 	 *
-	 * @return string
+	 * @return void
 	 */
 	public function setAppValue($key, $value) {
-		return $this->config->setAppValue($this->appName, $key, $value);
+		$this->validateValue($key, $value);
+		$this->config->setAppValue($this->appName, $key, $value);
+	}
+
+	/**
+	 * @param string $key
+	 * @param string $value
+	 *
+	 * @return void
+	 */
+	public function validateValue($key, $value) {
+		if (
+			$key === 'av_mode'
+			&& $value === 'icap'
```

Comment:
Quick style check: I'm more used to see the `&&` at the end of the condition. You can keep this if there are other places with the same style, so up to you.